tool/src/org/antlr/v4/semantics/SemanticPipeline.java

```diff
@@ -68,6 +69,11 @@
  *  tokens and rules from the imported grammars into a single collection.
  */
 public class SemanticPipeline {
+	protected final Set<String> reservedNames = new HashSet<String>();
```

Comment:
:bulb: This should be `static` if possible. Also the typical naming for a Java constant of this form would be `RESERVED_NAMES`.